internal/storage/google_cloud_storage_test.go

```diff
@@ -0,0 +1,223 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"crypto/sha256"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"strconv"
+	"strings"
+	"testing"
+
+	"cloud.google.com/go/storage"
+)
+
+func testGoogleCloudStorageClient(tb testing.TB) *storage.Client {
+	tb.Helper()
+
+	ctx := context.Background()
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		tb.Fatal(err)
+	}
+	return client
+}
+
+func testName(tb testing.TB) string {
+	tb.Helper()
+
+	var b [512]byte
+	if _, err := rand.Read(b[:]); err != nil {
+		tb.Fatalf("failed to generate random: %v", err)
+	}
+	digest := fmt.Sprintf("%x", sha256.Sum256(b[:]))
+	return digest[:32]
+}
+
+func testGoogleCloudStorageBucket(tb testing.TB) string {
+	tb.Helper()
+
+	if testing.Short() {
+		tb.Skipf("🚧 Skipping Google Cloud Storage tests (short!")
+	}
+
+	if skip, _ := strconv.ParseBool(os.Getenv("SKIP_GOOGLE_CLOUD_STORAGE_TESTS")); skip {
+		tb.Skipf("🚧 Skipping Google Cloud Storage tests (SKIP_GOOGLE_CLOUD_STORAGE_TESTS is set)!")
+	}
+
+	bucketID := os.Getenv("GOOGLE_CLOUD_BUCKET")
+	if bucketID == "" {
+		tb.Fatal("missing GOOGLE_CLOUD_BUCKET!")
```

Comment:
Do we need a corresponding change to docs/dev env to have this configured?

Reply:
Yea, but I'm not sure what that change looks like until we get more information on how to give Prow a service account securely.